proxy/attempt-register-vm-on-proxy.sh

```diff
@@ -35,15 +35,6 @@ function run-proxy-agent {
         --health-check-unhealthy-threshold=${HEALTH_CHECK_UNHEALTHY_THRESHOLD}
 }
 
-# Check if the cluster already have proxy agent installed by checking ConfigMap.
-if kubectl get configmap inverse-proxy-config; then
```

Comment:
Removing this check will result in URL change if the proxy pod is restarted for any reason. 
I think what you could do instead is check whether BACKEND_ID/PROXY_URL is empty or not.

Reply:
this PR is based on https://github.com/kubeflow/pipelines/pull/2391
Please review that one first.

In this PR, it only change the deployer Dockerfile.
I will remove the [WIP] from the title after PR2391 got merged.

Reply:
By the way, to your question of changed URL, it won't. it uses VM_ID as part of key to query/create the URL. Let's continue the comments in #2391.